viewing_party/party.py

```diff
@@ -1,23 +1,224 @@
+movies_dict = {}
+
+user_data = {"watched": [],
+            "watchlist" : []}
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    
+    movies_dict["title"] = title
+    movies_dict["genre"] = genre
+    movies_dict["rating"] = rating
+
+    return movies_dict
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie_info in user_data["watchlist"]:
+        if movie_info["title"] == title:
+            user_data["watchlist"].remove(movie_info)
+            user_data["watched"].append(movie_info)
+    
+    return user_data
+
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
+
+def get_watched_avg_rating(user_data):
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    
+    rating_total = 0
+    for movie in user_data["watched"]:
+        rating_total += movie["rating"]
+    average = rating_total / len(user_data["watched"])
+    return average
+    
+def get_most_watched_genre(user_data):
+
+    if len(user_data["watched"]) == 0.0:
+        return None
+    genre_count = {}

```

Comment:
Since you are checking the `len` of an object it will only be an integer (in this case specifically `0`). 
```suggestion
    if len(user_data["watched"]) == 0:
        return None
        
    genre_count = {}

```